File: edutask-core/src/main/java/ru/usernamedrew/edutaskcore/entity/TestCase.java
package ru.usernamedrew.edutaskcore.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "test_case")
public class TestCase extends BaseEntity {
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "task_id", nullable = false)
    private Task task;

    @Column(name = "input_data", nullable = false, columnDefinition = "text")
    private String inputData;

    @Column(name = "expected_output", nullable = false, columnDefinition = "text")
    private String expectedOutput;

    @Column(name = "is_hidden", nullable = false)
    private boolean hidden = false;

    @Column(nullable = false)
    private int points = 0;
}

File: edutask-frontend/src/types/testCase.ts
export type TestCaseResponse = {
  id: string;
  taskId: string;
  inputData: string;
  expectedOutput: string;
  hidden: boolean;
  points: number;
  createdAt: string;
  updatedAt: string;
};

export type TestCaseCreateRequest = {
  inputData: string;
  expectedOutput: string;
  hidden?: boolean;
  points?: number;
};

export type TestCaseUpdateRequest = Partial<TestCaseCreateRequest>;
